mailyak.go

```diff
@@ -34,47 +43,86 @@ type MailYak struct {
 const mailDateFormat = time.RFC1123Z
 
 // New returns an instance of MailYak using host as the SMTP server, and
-// authenticating with auth where required.
+// authenticating with auth.
 //
 // host must include the port number (i.e. "smtp.itsallbroken.com:25")
 //
-// 		mail := mailyak.New("smtp.itsallbroken.com:25", smtp.PlainAuth(
-// 			"",
-// 			"username",
-// 			"password",
-// 			"stmp.itsallbroken.com",
-//		))
+//      mail := mailyak.New("smtp.itsallbroken.com:25", smtp.PlainAuth(
+//          "",
+//          "username",
+//          "password",
+//          "smtp.itsallbroken.com",
+//      ))
 //
+// MailYak instances created with New will switch to using TLS after connecting
+// if the remote host supports the STARTTLS command. For an explicit TLS
+// connection, or to provide a custom tls.Config, use NewWithTLS() instead.
 func New(host string, auth smtp.Auth) *MailYak {
 	return &MailYak{
 		headers:        map[string]string{},
 		host:           host,
 		auth:           auth,
+		sender:         newSenderWithStartTLS(),
 		trimRegex:      regexp.MustCompile("\r?\n"),
 		writeBccHeader: false,
 		date:           time.Now().Format(mailDateFormat),
 	}
 }
 
+// NewWithTLS returns an instance of MailYak using host as the SMTP server over
+// an explicit TLS connection, and authenticating with auth.
+//
+// host must include the port number (i.e. "smtp.itsallbroken.com:25")
+//
+//      mail := mailyak.NewWithTLS("smtp.itsallbroken.com:25", smtp.PlainAuth(
+//          "",
+//          "username",
+//          "password",
+//          "smtp.itsallbroken.com",
+//      ), tlsConfig)
+//
+// If tlsConfig is nil, a sensible default is generated that can connect to
+// host.
+func NewWithTLS(host string, auth smtp.Auth, tlsConfig *tls.Config) (*MailYak, error) {
+	// If there is no TLS config provided, initialise a default.
+	if tlsConfig == nil {
+		// Split the hostname from the addr.
+		//
+		// This hostname is used during TLS negotiation and during SMTP
+		// authentication.
+		hostName, _, err := net.SplitHostPort(host)
+		if err != nil {
+			return nil, err
+		}
+
+		//nolint:gosec // Maximum compatability but please use TLS >= 1.2
+		tlsConfig = &tls.Config{
+			ServerName: hostName,
+		}
+	}
+
+	// Construct a default MailYak instance
+	m := New(host, auth)
+
+	// Initialise the TLS sender with the (potentially nil) TLS config,
```

Comment:
slightly misleading comment, as it's not nil at this point, and the `newSenderWithExplicitTLS` method requires that it be non nil

Reply:
Great catch - I wound up refactoring this and it got left behind

Thanks!